specification/cognitiveservices/data-plane/QnAMaker/preview/v5.0-preview.2/QnAMaker.json

```diff
@@ -0,0 +1,2082 @@
+    },
+    "KbId": {
+      "type": "string",
+      "name": "kbId",
+      "in": "path",
+      "required": true,
+      "x-nullable": false,
+      "description": "Knowledgebase id.",
+      "x-ms-parameter-location": "method"
+    },
+    "OperationId": {
+      "type": "string",
+      "name": "operationId",
+      "in": "path",
+      "required": true,
+      "x-nullable": false,
+      "description": "Operation id.",
+      "x-ms-parameter-location": "method"
+    },
+    "WordAlterations": {
+      "name": "wordAlterations",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/WordAlterationsDTO"
+      },
+      "x-nullable": true,
+      "description": "New alterations data.",
+      "x-ms-parameter-location": "method"
+    },
+    "KeyType": {
+      "type": "string",
+      "name": "keyType",
+      "in": "path",
+      "required": true,
+      "x-nullable": false,
+      "description": "Type of Key",
+      "x-ms-parameter-location": "method"
+    },
+    "GenerateAnswerPayload": {
+      "name": "generateAnswerPayload",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/QueryDTO"
+      },
+      "x-nullable": true,
+      "description": "Post body of the request.",
+      "x-ms-parameter-location": "method"
+    },
+    "TrainPayload": {
+      "name": "trainPayload",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/FeedbackRecordsDTO"
+      },
+      "x-nullable": true,
+      "description": "Post body of the request.",
+      "x-ms-parameter-location": "method"
+    },
+    "Endpoint": {
+      "name": "Endpoint",
+      "description": "Supported Cognitive Services endpoint (e.g., https://< qnamaker-resource-name >.api.cognitiveservices.azure.com).",
+      "x-ms-parameter-location": "client",
+      "required": true,
+      "type": "string",
+      "in": "path",
+      "x-ms-skip-url-encoding": true
+    },
+    "EndpointSettingsPayload": {
+      "name": "endpointSettingsPayload",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/EndpointSettingsDTO"
+      },
+      "x-nullable": true,
+      "description": "Post body of the request.",
+      "x-ms-parameter-location": "method"
+    },
+    "PrebuiltGenerateAnswerPayload": {
+      "name": "prebuiltGenerateAnswerPayload",
+      "in": "body",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/PrebuiltQuery"
+      },
+      "x-nullable": true,
+      "description": "Post body of the request.",
+      "x-ms-parameter-location": "method"
+    },
+    "StringIndexType": {
+      "name": "stringIndexType",
+      "in": "query",
+      "type": "string",
+      "description": "(Optional) Specifies the method used to interpret string offsets.  Defaults to Text Elements (Graphemes) according to Unicode v8.0.0.",
+      "default": "TextElements_v8",
+      "enum": [
```

Comment:
This implies a non-extensible enum.  Please ack if this is by design.

Reply:
Yes this is by design. This is also inline with the pattern followed by Text Analytics.